[rimgik] iP

rimgik opened the pull request:
# Nimbus
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Nimbus frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

rimgik committed:
Week 2: Level-0

rimgik committed:
Week 2: Level-1

rimgik committed:
Week 2: Level-2

rimgik committed:
Week 2: Level-3

rimgik committed:
Week 2: Level-4

rimgik committed:
Added test cases

rimgik committed:
Added error handling

rimgik committed:
Added support for removal of task

rimgik committed:
Add Enum for each command

rimgik committed:
Improve implementation of processCommand

rimgik committed:
Add support for saving file

rimgik committed:
Update .gitignore

rimgik committed:
Merge branch 'branch-Level-7'

rimgik committed:
Implement constructor chaining to avoid code duplication

rimgik committed:
Add support for datetime parsing

rimgik committed:
Merge branch 'branch-Level-8'

rimgik committed:
Add Class Ui

rimgik committed:
Move readSavedFile() from Nimbus to Storage Class

rimgik committed:
Finish implementation of Storage class

rimgik committed:
Organise class into package

rimgik committed:
Finish Parser class

rimgik committed:
Finish TaskList class

rimgik committed:
Merge branch 'branch-A-MoreOOP'

rimgik committed:
Merge branch 'add-gradle-support-local'

rimgik committed:
Add Junit test

rimgik committed:
Update gradle settings for shadowJar

rimgik committed:
Add header comment for method

rimgik committed:
Fix java coding violation

rimgik committed:
Add Find command

rimgik committed:
Refactor package name

rimgik committed:
Merge branch 'branch-A-JavaDoc'

rimgik committed:
Merge branch 'branch-A-CodingStandard'

rimgik committed:
Merge branch 'branch-Level-9'

rimgik committed:
Remove less frequently function from UI class

rimgik committed:
Fix coding standard violation

rimgik committed:
Add gui for the chatbot

rimgik committed:
Merge branch 'branch-Level-10'

rimgik committed:
Add CheckStyle to project

rimgik committed:
Merge branch 'branch-A-CheckStyle'

rimgik committed:
Fix all Java coding standard violation

rimgik committed:
Bug fix: Junit test not working

naythee169 commented on src/main/java/com/nimbus/Parser.java:
```diff
@@ -0,0 +1,106 @@
+     * @throws InvalidArgumentException if the argument of a command does not match its command type
+     */
+    public static Command parse(String line) throws InvalidCommandException, InvalidArgumentException {
+        return switch (getCommandType(line)) {
```
very good! switch-case is correctly formatted

naythee169 commented on src/main/java/com/nimbus/Todo.java:
```diff
@@ -0,0 +1,36 @@
+package com.nimbus;
+
+/**
+ * Todo is a task
```
May want to add a bit more elaboration?

Droas590 commented on src/main/java/com/commands/Command.java:
```diff
@@ -0,0 +1,21 @@
+/**
+ * Command to be executed by chatbot
+ */
+public abstract class Command {
```
I like the abstraction of this class.

Droas590 commented on src/main/java/com/nimbus/Task.java:
```diff
@@ -0,0 +1,75 @@
+/**
+ * Task to be done
+ */
+public abstract class Task {
```
Maybe can build another Task package with Todo, Event and Deadline?

Droas590 commented on src/main/java/com/nimbus/InvalidCommandException.java:
```diff
@@ -0,0 +1,10 @@
+/**
+ * Invalid command error for Nimbus
+ */
+public class InvalidCommandException extends Exception {
```
Maybe can abstract out to another package exception with other exceptions?

Droas590 commented on src/main/java/com/nimbus/Storage.java:
```diff
@@ -0,0 +1,143 @@
+    public static Task getTaskFromSavedCommand(String command) {
+        String[] arr = command.split("\\|");
+        boolean isDone = arr[1].equals("1");
+        return switch (arr[0]) {
```
I like this switch statement you are using. Very clean.

Droas590 reviewed (commented):
LGTM. Overall great code quality, I can see the author put a lot of effort on this ip.

naythee169 commented on src/main/java/com/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,33 @@
+    public void execute(Ui ui, Storage storage, TaskList tasks)
+            throws InvalidArgumentException {
+        Task task = new Deadline(getDescription(argument),
+                readOption(argument, "by"));
```
perhaps add javadocs explaining what can option is?

naythee169 commented on src/main/java/com/nimbus/Ui.java:
```diff
@@ -0,0 +1,134 @@
+    /**
+     * Print welcome message onto window
+     */
+    public void showWelcomeMessage() {
```
maybe put the different messages into another class? inside the Ui package

naythee169 commented on src/main/java/com/nimbus/InvalidCommandException.java:
```diff
@@ -0,0 +1,10 @@
+package com.nimbus;
```
May want to consider putting the exceptions in their own package

naythee169 reviewed (commented):
Good job! may want to consider having more packages such as an exceptions package and a parser package etc.

rimgik committed:
Add some assertion to file

rimgik committed:
Improve Code Quality for some file

There are several code quality issues across multiple file that need to
be fixed.

We fix all those file to improve code quality.

rimgik committed:
Merge pull request #1 from rimgik/branch-A-Assertions

Add some assertion to file

rimgik committed:
Merge branch 'master' into branch-A-CodeQuality

rimgik committed:
Merge pull request #2 from rimgik/branch-A-CodeQuality

Improve code quality for some file

rimgik committed:
Find command shouldn't search for keyword in date

rimgik committed:
Add detection support for duplicate task

rimgik committed:
Add CI workflow to .github/workflows

rimgik committed:
Fix coding stadard violation

rimgik committed:
Update build.gradle

rimgik committed:
Update several JavaDoc header comment to follow coding standard

rimgik committed:
Update user guide under docs

rimgik committed:
Update user guide under docs

rimgik committed:
Fix showFoundTask() not printing newline after each task

rimgik committed:
Update image for the user guide

rimgik committed:
Update build.gradle for shadowJar

rimgik committed:
Remove unnecessary dependencies from build.gradle
